pkg/descheduler/strategies/pod_lifetime.go

```diff
@@ -84,7 +93,14 @@ func PodLifeTime(ctx context.Context, client clientset.Interface, strategy api.D
 				if string(pod.Status.Phase) == phase {
```

Comment:
From https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase:
```
The number and meanings of Pod phase values are tightly guarded.
Other than what is documented here, nothing should be assumed
about Pods that have a given phase value.
```

We use `pod status phase` term which might be easy to confuse with `pod phase`. Also, the upstream documentation mentions `container status` with three possible values: `Waiting`, `Running`, and `Terminated`.

With `PodInitializing` and `ContainerCreating` as the pod status phase values it will get even more confusing what is what. Although, I agree it's practical and preferable to have a single field for all the values.

If anything, we need to be explicit about what the field is used for in https://github.com/kubernetes-sigs/descheduler#podlifetime. Though, I prefer to use more generic word "state" rather than "status" if the list of values gets extended with `PodInitializing` and `ContainerCreating`. Worth adding a new field and deprecating `podStatusPhases`.

CC @knelasevero

Reply:
That makes sense to me.

If we have a consensus, then I can add the new field.

Reply:
Even if it is a bit confusing, wouldn't it be better to take advantage of the version being v1alpha1 and deprecate "before it's too late"? Alpha versions are for that, right?

Reply:
There is a second [commit](https://github.com/kubernetes-sigs/descheduler/pull/834/commits/446dd5a11fb7c3927ec9fb4090d7511c8e2b2e6b) in this PR that adds the new field `States` and deprecates `PodStatusPhases`

Reply:
ping